sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1516,3 +1517,51 @@ def test_multi_core_gridsearch_and_early_stopping():
                                 random_state=0)
     search.fit(iris.data, iris.target)
     assert search.best_score_ > 0.8
+
+
+@pytest.mark.skipif(
+        not hasattr(sp, "random"),
+        reason="this test uses scipy.random, that was introduced in version  "
+        "0.17. This skip condition can be dropped as soon as we drop support "
+        "for scipy versions older than 0.17")
+@pytest.mark.parametrize("backend",
+                         ["loky", "multiprocessing", "threading"])
+def test_SGDClassifier_fit_for_all_backends(backend):
+    # This is a non-regression smoke test. In the multi-class case,
+    # SGDClassifier.fit fits each class in a one-versus-all fashion using
+    # joblib.Parallel.  However, each OvA step updates the coef_ attribute of
+    # the estimator in-place. Internally, SGDClassifier calls Parallel using
+    # require='sharedmem'. This test makes sure SGDClassifier.fit works
+    # consistently even when the user asks for a backend that does not provide
+    # sharedmem semantics.
+
+    # We further test a case where memmapping would have been used if
+    # SGDClassifier.fit was called from a loky or multiprocessing backend. In
+    # this specific case, in-place modification of clf.coef_ would have caused
+    # a segmentation fault when trying to write in a readonly memory mapped
+    # buffer.
+
+    random_state = 42
```

Comment:
If I use this I cannot use `random_state=random_state` in the classifier initialization, as both fitting processes need to be started with the exact same random state, and this will mutate it in between.

Reply:
Then just pass 42 directly, and the code will be more readable for it